projects/novo-elements/src/elements/data-table/cell-headers/data-table-checkbox-header-cell.component.ts

```diff
@@ -70,6 +74,19 @@ export class NovoDataTableCheckboxHeaderCell<T> extends CdkHeaderCell implements
   }
 
   public onClick(): void {
-    this.dataTable.selectRows(!this.checked);
+    if (this.isAtLimit()) {
+      this.toaster.alert({
+        theme: 'danger',
+        position: 'fixedTop',
+        message: 'Error, more than 500 files are not able to be selected at one time',
+        icon: 'caution',
+      });
+    } else {
+      this.dataTable.selectRows(!this.checked);
+    }
+  }
+
+  public isAtLimit() {
+    return this.dataTable.state.selectedRows.size + this.dataTable.state.pageSize >= 500;
```

Comment:
might want to convert this to a getter for a variable since you're calling this from the template